test/models/robot_directory_test.rb

```diff
@@ -0,0 +1,147 @@
+                           state: 'Colorado',
+                           birthdate: '09/09/09',
+                           date_hired: '10/10/10',
+                           department: 'Manager')
+
+    assert_equal 2, robot_directory.all.count
+    assert_equal current_robot_id, robot_directory.all.first.id
+    assert_instance_of Robot, robot_directory.all[1]
+  end
+
+  def test_it_finds_robots
+    robot_directory.create(name: 'George',
+                           city: 'Denver',
+                           state: 'Colorado',
+                           birthdate: '10/10/10',
+                           date_hired: '11/11/11',
+                           department: 'Loader')
+
+    robot_directory.create(name: 'Sandy',
+                           city: 'Littleton',
+                           state: 'Colorado',
+                           birthdate: '09/09/09',
+                           date_hired: '10/10/10',
+                           department: 'Manager')
+
+    robot = robot_directory.find(current_robot_id)
+    assert_equal "George", robot.name
+
+    robot = robot_directory.find(current_robot_id + 1)
+    assert_equal "Sandy", robot.name
+  end
+
+  def test_it_updates
+    robot_directory.create(name: 'George',
+                           city: 'Denver',
+                           state: 'Colorado',
+                           birthdate: '10/10/10',
+                           date_hired: '11/11/11',
+                           department: 'Loader')
+
+    updated_robot = {name: 'George',
+                     city: 'Denver',
+                     state: 'Colorado',
+                     birthdate: '10/10/10',
+                     date_hired: '11/11/11',
+                     department: 'CEO'}
+
+    robot = robot_directory.find(current_robot_id)
+    assert_equal "Loader", robot.department
+
+    robot_directory.update(robot.id, updated_robot)
+
+    robot = robot_directory.find(current_robot_id)
+    assert_equal "CEO", robot.department
+  end
+
+  def test_it_destroys
+    assert_equal 0, robot_directory.all.count
+
+    robot_directory.create(name: 'George',
+                           city: 'Denver',
+                           state: 'Colorado',
+                           birthdate: '10/10/10',
+                           date_hired: '11/11/11',
+                           department: 'Loader')
+
+    robot_directory.create(name: 'Sandy',
+                           city: 'Littleton',
+                           state: 'Colorado',
+                           birthdate: '09/09/09',
+                           date_hired: '10/10/10',
+                           department: 'Manager')
+
+    assert_equal 2, robot_directory.all.count
+    robot = robot_directory.find(current_robot_id)
+
+    robot_directory.destroy(robot.id)
+
+    assert_equal 1, robot_directory.all.count
+  end
+
+  def test_the_delete_all_method
+    assert_equal 0, robot_directory.all.count
+
+    robot_directory.create(name: 'George',
```

Comment:
this line exists in quite a few different tests in this file. it's probably a good idea to create a method that does this for you so you don't have to re-write it over and over 👍